### PR TITLE
Add deterministicJS

### PR DESCRIPTION
**Pull request by bebef1987**
New Deterministic JS

### CODE REVIEW

File: mitmproxy.py [hunk 1]
```diff
@@ -23,7 +23,16 @@ def binary(self):
...
         if self.record:
-            command = [self.binary, "--wfile", self.path]
+            command = [
+                self.binary,
+                "--wfile",
+                self.path,
+                "--script",
+                " ".join(
+                    [
+                        os.path.join(self.scripts, "inject-deterministic.py")]
```
Comment (davehunt) [inline mitmproxy.py:33]: Does this also need to be used when replaying, or does it just impact the recordings?

File: scripts/deterministic.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (davehunt) [inline scripts/deterministic.js]: As this is taken from https://github.com/catapult-project/catapult/blob/183d99e38bfe9d296913b357c80599577056700f/web_page_replay_go/deterministic.js we should include the copyright statement and license. I would suggest creating a subdirectory named 'catapult' and putting the LICENSE file from https://github.com/catapult-project/catapult/blob/master/LICENSE in there.

File: scripts/inject-deterministic.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                        js = jsfile.read().replace("REPLACE_LOAD_TIMESTAMP", str(millis))
+                        new_html = html[:firstScriptIndex] + "<script>" + js + "</script>" + html[firstScriptIndex:]
+
+                        flow.response.content = bytes(new_html, "utf-8")
+                    ctx.log.info("In request {} added deterministic JS".format(flow.request.url))
+                else:
+                    ctx.log.info("No script tags found in request {}".format(flow.request.url))
+
+def start():
+    ctx.log.info("Load Detrministic JS")
```
Comment (davehunt) [inline scripts/inject-deterministic.py]: typo "Deterministic"

File: scripts/inject-deterministic.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import, print_function
+
+
+import time
+from mitmproxy import ctx
+
+
+class AddDetrministic():
```
Comment (davehunt) [inline scripts/inject-deterministic.py]: typo "AddDeterministic"

**Review by davehunt — CHANGES_REQUESTED** [review]

File: mitmproxy.py (hunk 1 above)
Comment (bebef1987) [reply mitmproxy.py:33]: This only impacts recording. it should not be needed for replay

File: scripts/deterministic.js (hunk 2 above)
Comment (bebef1987) [reply scripts/deterministic.js]: Done

File: scripts/inject-deterministic.py (hunk 3 above)
Comment (bebef1987) [reply scripts/inject-deterministic.py]: Done

File: scripts/inject-deterministic.py (hunk 4 above)
Comment (bebef1987) [reply scripts/inject-deterministic.py]: Done

**Review by davehunt — APPROVED** [review]

**Pull request merged**